File: LAb_1/Tham lam/trade_money.py
# Give 5 types of coin 1, 5, 10, 25, 100 cent
# find the way to give X cent which the number of coins 
# is minimun using Greedy Solution



# USING GREEDY


X = int(input())
Coins = [100, 25, 10, 5, 1]
Trade = []
for i in Coins:
    if (X / i >=1):
        Trade.append(int(X/i))
        X = X - i* int(X/i)
    else: Trade.append(0)

print(Trade)

File: LAb_1/Tham lam/trade_money.cpp
#include<iostream>
#include<string>
using namespace std;

// USING GREEDY


int main()
{   int coins[5] = {100,25,10,5,1};
    int X;
    string Trade = "";
    cin>>X;
    for (int i = 0;i<5;i++){
        if (X/coins[i] >=1){
            Trade +=to_string(int(X/coins[i]));
            X -= coins[i]*int(X/coins[i]);
        }
        else {
            Trade +="0";
        }
    }
    cout<<Trade<<endl;

    return 0;
}




    
